avocado/utils/memory.py

```diff
@@ -278,3 +279,35 @@ def get_buddy_info(chunk_sizes, nodes="all", zones="all"):
             buddyinfo_dict[chunk_size] += int(chunk_info)
 
     return buddyinfo_dict
+
+
+def set_thp_value(value, feature):
+    """
+    Sets THP feature to a given value
+
+    :param value : Value to be set to feature
+    :type value : str
+    :param feature : Thp feature to set
+    :type feature : str
+    """
+    thp_path = '/sys/kernel/mm/transparent_hugepage/'
+    thp_feature_to_set = os.path.join(thp_path, feature)
+    cmd = 'echo %s > %s ' % (value, thp_feature_to_set)
```

Comment:
If invalid arguments are passed from test case like,
memory.set_thp_value('always', 'khugepaged/defrag')
Then, it gives out error like, ' IOError: [Errno 22] Invalid argument '
But if process.system is used we can get error message directly that the command has failed like below,
CmdError: Command 'echo always > /sys/kernel/mm/transparent_hugepage/khugepaged/defrag ' failed (rc=1)
So, which is the better way to indicate as a failure??